packages/jest-cli/src/runGlobalHook.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+import type {GlobalConfig} from 'types/Config';
+import type {Test} from 'types/TestRunner';
+
+export default async ({
+  allTests,
+  globalConfig,
+  moduleName,
+}: {
+  allTests: Array<Test>,
+  globalConfig: GlobalConfig,
+  moduleName: string,
+}) => {
+  const globalModulePaths = new Set(
+    allTests.map(test => test.context.config[moduleName]),
+  );
+
+  if (globalConfig[moduleName]) {
+    globalModulePaths.add(globalConfig[moduleName]);
+  }
+
+  if (globalModulePaths.size > 0) {
+    await Promise.all(
+      Array.from(globalModulePaths).map(async modulePath => {
+        if (!modulePath) {
+          return null;
```

Comment:
any scenario where this could happen?

Reply:
I think I did it because `globalSetup`/`globalTeardown` are nullables and can also be an empty string.

https://github.com/facebook/jest/blob/c1d3de5e315d4dc01acfe33abd7de4fe7d4591ea/types/Config.js#L120-L121

I did not write any specific test for this use case, but Jest's tests would not pass without this condition.